src/spox/_node.py

```diff
@@ -130,7 +126,7 @@ def __init__(
             # As inference functions may access which output vars we initialized (e.g. variadics)
             # we inject uninitialized vars first
             self.outputs = self._init_output_vars()
-            self.inference(infer_types, propagate_values)
+            self.inference(infer_types, input_prop_values)
```

Comment:
I would find it conceptually easier if we could separate the value propagation from the initialization of the `Node` object. Do you have an example where we may need the `input_prop_values` at this stage?